src/js/components/stock-movement-wizard/request/AddItemsPage.jsx

```diff
@@ -879,27 +595,39 @@ class AddItemsPage extends Component {
     const { data } = response.data;
     let lineItemsData;
 
+    const isPullType = _.get(this.state.values.replenishmentType, 'name') === REPLENISHMENT_TYPE_PULL;
+
     if (this.state.values.lineItems.length === 0 && !data.length) {
       lineItemsData = new Array(1).fill({ sortOrder: 100 });
     } else if (this.state.isRequestFromWard && _.get(this.state.values.stocklist, 'id')) {
       lineItemsData = _.map(
         data,
         (val) => {
-          const { quantityRequested, demandPerReplenishmentPeriod, quantityOnHand } = val;
-          const qtyRequested =
-              quantityRequested || demandPerReplenishmentPeriod - quantityOnHand;
+          const {
+            quantityRequested,
+            demandPerReplenishmentPeriod,
+            quantityOnHand,
+            quantityAllowed,
+          } = val;
+
+          let qtyRequested = 0;
+          if (qtyRequested) qtyRequested = quantityRequested;
```

Comment:
This part is weird. Will it be ever entered?
Why you can't do a simple:
` const qtyRequested = isPullType ? (demandPerReplenishmentPeriod - quantityOnHand) : quantityAllowed`?

Reply:
no, it seems that it will never enter, this is a mistake from my part.
Instead this is supposed to be:

```js
  let qtyRequested = 0;
  if (quantityRequested) qtyRequested = quantityRequested;
  else if (isPullType) qtyRequested = demandPerReplenishmentPeriod - quantityOnHand;
  else qtyRequested = quantityAllowed;
```

since previously there was a _two way_ condition  
```js
  const qtyRequested =
      quantityRequested || demandPerReplenishmentPeriod - quantityOnHand;
```
and now I also need to set **qtyRequested** to **quantityAllowed** if _request type_ is **Push**